templates/post/fileinfo.html

```diff
@@ -20,12 +20,13 @@
 				{% endif %}
 			{% endif %}
 			{% if config.show_filename and file.filename %}
-				, 
+				, <span class="postfilename"><a href="{{ config.uri_img }}{{ file.file|e|bidi_cleanup }}" download="{{ file.filename|e|bidi_cleanup }}"
 				{% if file.filename|length > config.max_filename_display %}
-					<span class="postfilename" title="{{ file.filename|e|bidi_cleanup }}">{{ file.filename|truncate_filename(config.max_filename_display)|e|bidi_cleanup }}</span>
+					title="Save as original filename: {{ file.filename|e|bidi_cleanup }}">{{ file.filename|truncate_filename(config.max_filename_display)|e|bidi_cleanup }}
 				{% else %}
-					<span class="postfilename">{{ file.filename|e|bidi_cleanup }}</span>
+					title="Save as original filename">{{ file.filename|e|bidi_cleanup }}
```

Comment:
I don't understand why you'd not show the original filename in the `title` if it's not a long filename. Also, moving that out of the branch as well allows you to be more DRY.

> [Also I really think my role with this project is just teaching people how to code better these days lol.](https://github.com/vichan-devel/vichan/issues/525#issuecomment-1379972259)